wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemRam/DockerBasedSystemRamLimitChecker.java

```diff
@@ -0,0 +1,141 @@
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.systemRam;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.Inject;
+
+import org.eclipse.che.commons.lang.Size;
+import org.eclipse.che.commons.schedule.ScheduleDelay;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.SystemInfo;
+import org.everrest.core.impl.provider.json.JsonUtils;
+import org.everrest.websockets.WSConnectionContext;
+import org.everrest.websockets.message.ChannelBroadcastMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import javax.websocket.EncodeException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import static com.codenvy.service.systemRam.SystemRamLimitStatus.SYSTEM_RAM_LIMIT_EXCEEDED;
+import static com.codenvy.service.systemRam.SystemRamLimitStatus.SYSTEM_RAM_LIMIT_NOT_EXCEEDED;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Implementation of {@link SystemRamLimitChecker}
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class DockerBasedSystemRamLimitChecker implements SystemRamLimitChecker{
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerBasedSystemRamLimitChecker.class);
+
+    private final DockerConnector dockerConnector;
+    private final ExecutorService executor;
+
+    private volatile boolean systemRamLimitExceedMessageSent;
+    private volatile SystemRamLimitStatus limitStatus = SYSTEM_RAM_LIMIT_NOT_EXCEEDED;
+
+    @Inject
+    public DockerBasedSystemRamLimitChecker(DockerConnector dockerConnector) {
+        this.dockerConnector = dockerConnector;
+        executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                                                         .setNameFormat("DockerApiConnector-%d")
+                                                         .setDaemon(true)
+                                                         .build());
+    }
+
+    @Override
+    public final SystemRamLimitStatus getSystemRamLimitStatus() {
+        long systemRamUsed = 0;
+        long systemRamTotal = 0;
+        List<String> allNodesRamUsage = new ArrayList<>();
+        Callable<SystemInfo> task = dockerConnector::getSystemInfo;
+        Future<SystemInfo> future = executor.submit(task);
+        SystemInfo systemInfo;
+        try {
+            systemInfo = future.get(5, SECONDS);
```

Comment:
There is no default timeout in this request. Theoretically `dockerConnector.getSystemInfo()` can freeze without timeout